.github/workflows/publish.yml

```
@@ -63,8 +63,14 @@ jobs:
           GH_TOKEN: ${{ github.token }}
         working-directory: src/D2L.Bmx/build
         run: |
-          Compress-Archive -Path "$env:FILE_NAME" -DestinationPath "bmx-$env:PLATFORM.zip"
-          gh release upload "$env:RELEASE_TAG" "bmx-$env:PLATFORM.zip"
+          if ($env:PLATFORM -eq 'win') {
+            Compress-Archive -Path "bmx.exe" -DestinationPath "bmx-win.zip"
+            gh release upload "$env:RELEASE_TAG" "bmx-win.zip"
+          } else if ($env:PLATFORM -eq 'osx') {
+            chmod +x "bmx"
+            tar -czvf "bmx-osx.tar.gz" "bmx"
+            gh release upload "$RELEASE_TAG" "bmx-osx.tar.gz"
```

you're still in powershell here
```suggestion
            gh release upload "$env:RELEASE_TAG" "bmx-osx.tar.gz"
```

---

there's also the option of doing the `if` at the GitHub Actions step level, if you want to use different shells for the two conditions
https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idstepsif